src/main/java/org/humancellatlas/ingest/state/persistence/AutoLoader.java

```diff
@@ -29,23 +42,113 @@
+                UUID envelopeUuid = UUID.fromString(stateMachine.getId());
+
+                Optional<SubmissionEnvelopeReference> envelope = getEnvelope(envelopeUuid);
+
+                sync(stateMachine, envelope);
+
+                if (envelope.isPresent()) {
+                    submissionStateMonitor.monitorSubmissionEnvelope(envelope.get(), stateMachine);
+
+                    String state = stateMachine.getState().getId().toString();
+
+                    log.info(String.format("Restored %s with current state: %s", envelopeUuid.toString(), state));
+                }
+
+
             } catch (RuntimeException e) {
-                log.error(String.format("Failed to autoload statemachine for envelope with UUID: %s", envelopeUuid.toString()), e);
+                String desc = String.format("An unexpected error has occurred in loading the state machine for submission %s: %s",
+                        stateMachine.getId(), e.getMessage());
+
+                String action = "Please check the stack trace.";
+
+                handleException(desc, action, e);
             }
         });
     }
 
+    private Optional<SubmissionEnvelopeReference> getEnvelope(UUID envelopeUuid) {
+        try {
+            SubmissionEnvelopeReference envelope = ingestApiClient.referenceForSubmissionEnvelope(envelopeUuid);
+            return Optional.of(envelope);
+        } catch (HttpClientErrorException.NotFound e) {
+            log.info(String.format("Submission %s was not found.", envelopeUuid.toString()));
+            return Optional.empty();
+        } catch (RestClientException e) {
+            String desc = String.format("An error has occurred in retrieving submission envelope %s : %s",
+                    envelopeUuid.toString(), e.getMessage());
+            String action = "Please ensure Ingest API is running.";
+            handleException(desc, action, e);
+        }
+        return Optional.empty();
+    }
+
+    private Optional<StateMachine<SubmissionState, SubmissionEvent>> sync(StateMachine<SubmissionState, SubmissionEvent> stateMachine, Optional<SubmissionEnvelopeReference> envelope) {
+        if (envelope.isEmpty()) {
+            deleteStateMachine(stateMachine);
+            return Optional.empty();
+        } else {
+            return Optional.of(updateState(stateMachine, envelope.get()));
+        }
+    }
+
+    private StateMachine<SubmissionState, SubmissionEvent> updateState(StateMachine<SubmissionState, SubmissionEvent> stateMachine, SubmissionEnvelopeReference envelope) throws UnrecognisedSubmissionStateException {
+        SubmissionState currentState = stateMachine.getState().getId();
+        SubmissionState correctState = convertState(envelope.getState());
+
+        // Note that some "ongoing" states like Draft/Validating/Invalid/Processing/Exporting can have extended states which
```

Comment:
What is the purpose of this comment?
How will anyone find it after this PR?
Is it something we should do now?
Should it go in a ticket instead?

Reply:
I believe my suggestion here is a big work, I was suggesting to remove the state tracker component and move the use of state machines inside core.

Reply:
The purpose of the comment is to note that the sync logic can only safely force set some statemachines to some specific states.

Reply:
Is there a pressing need for this at this moment?
If not, leave it be. If you think it is important but not urgent, you can open a ticket, and we might encounter it down the road.

Reply:
No, there's no pressing need at the moment.